draft-ietf-mls-protocol.md

```diff
@@ -1848,11 +1848,18 @@ enum {
     (255)
 } SenderType;
 
+struct {
+    uint8 KeyPackageID<0..255>;
+} MemberId
```

Comment:
Sounds like a good approach. I'll also note, that while we use a KeyPackage hash to identify members, these hashes change when the key package is updated, whereas the identity of the member does not. This could lead to errors, for example, when processing a `remove` right after an `update` for the same member was processed. I'm not entirely sure that's actually possible, but maybe you understand my concern. I don't want people to thing that KeyPackage hashes are static mappings to members.

Reply:
Good point.  And to make it worse, we have the following:

> Update proposals are applied first, followed by Remove proposals

Maybe we should just declare that a single Commit containing an Update and a Remove for the same leaf is invalid.

Reply:
That would make sense, since updating and then immediately removing is rather pointless. Alternatively, we could make the KeyPackageID relative to the tree of the last epoch.

Reply:
> In any case, I don't think this extra layer of `struct` wrapping is needed.

The advantage of keeping the struct layer is that the type is more expressive. If we don't have the struct, we end up with, for example, 

```
~~~~~
struct {
     opaque removed<0..255>;
} Remove;
~~~~~
```
without any indication what `removed` actually is.